R/mobr_boxplots.R

```diff
@@ -33,6 +33,130 @@ calc_PIE = function(x) {
     return(H)
 }
 
+# Calculate biodiversity indices of single group
+
+calc_biodiv_single_group <- function(abund_vec, n_rare){
+   
+   abund_vec <- abund_vec[abund_vec > 0]
+   
+   out_vec <- c("N"       = NA,
+                "S"       = NA,
+                "S_rare"  = NA,
+                "S_asymp" = NA,
+                "PIE"     = NA,
+                "ENS_PIE" = NA
+                )
+
+   if (length(abund_vec) > 0){
+      out_vec["N"] <- sum(abund_vec)
+      out_vec["S"] <- length(abund_vec)
+      if (!is.na(n_rare))
+         out_vec["S_rare"] <- rarefaction(abund_vec, method = "indiv", effort = n_rare)
+      out_vec["PIE"] <- diversity(abund_vec, index = "simpson")
+      out_vec["ENS_PIE"] <- diversity(abund_vec, index = "invsimpson")
+      
+      S_asymp <- try(vegan::estimateR(abund_vec))
+      if (class(S_asymp) == "try_error"){
+         warning("The Chao richness estimator cannot be calculated for all groups.")
+      } else {
+         out_vec["S_asymp"] = S_asymp["S.chao1"]
+      }
+   }
+   
+   return(out_vec)
+}
+
+# generate a single bootstrap sample of group-level biodiversity indices
+boot_sample_group <- function(comm_dat, n_rare)
+{
+   n_sample <- nrow(comm_dat)
+   sample_rows <- sample(1:n_sample, size = n_sample, replace = T)
+   abund <- colSums(comm_dat[sample_rows, ])
+   out <- calc_biodiv_single_group(abund, n_rare = n_rare)
+   
+   return(out)
+}
+
+# replicated bootstrap samples of group-level indices
+repl_boot_samples <- function(comm_dat, n_rare, n_perm = 100)
+{
+   repl_biodiv <- replicate(n_perm, boot_sample_group(comm_dat, n_rare = n_rare))
+   return(repl_biodiv)
+}
+
```

Comment:
I would vote to drop this function simply because it only uses one other function `replicate` which most R users should be familiar with. If you don't mind let's just drop this function and provide a call to `replicate` this will make our code easier to follow for readers.